doc/stdenv/platform-notes.chapter.md

```diff
@@ -59,79 +57,59 @@ Some common issues when packaging software for Darwin:
 
-  Note: Some packages may hardcode an absolute path to `xcrun`, `xcodebuild`, or `xcode-select`. Those paths should be removed or replaced.
+#### Package requires `xcodebuild`
+
+The xcbuild package provides an `xcodebuild` command for packages that really depend on Xcode. This replacement is not 100% compatible and may run into some issues, but it is able to build many packages. To use `xcodebuild`, add `xcbuildHook` to your package’s `nativeBuildInputs`. It will provide a `buildPhase` for your derivation. You can use `xcbuildFlags` to specify flags to `xcodebuild` such as the required schema. If a schema has spaces in its name, you must set `__structuredAttrs` to `true`. See MoltenVK for an example of setting up xcbuild.
 
   ```nix
   stdenv.mkDerivation {
     name = "libfoo-1.2.3";
-    prePatch = ''
-      substituteInPlace Makefile \
-        --replace-fail /usr/bin/xcrun xcrun
-        # or: --replace-fail /usr/bin/xcrun '${lib.getExe' buildPackages.xcbuild "xcrun"}'
-    '';
+    xcbuildFlags = [
+      "-configuration"
+      "Release"
+      "-project"
+      "libfoo-project.xcodeproj"
+      "-scheme"
+      "libfoo Package (macOS only)"
+    ];
+    __structuredAttrs = true;
   }
   ```
 
-- Multiple SDKs are available for use in nixpkgs. Each platform has a default SDK (10.12.2 for x86_64-darwin and 11.3 for aarch64-darwin), which is available as the `apple-sdk` package.
+#### Package requires a non-default SDK or fails to build due to missing frameworks or symbols
 
-  The SDK provides the necessary headers and text-based stubs to link common frameworks and libraries (such as libSystem, which is effectively Darwin’s libc). Projects will sometimes indicate which SDK to use by the Xcode version. As a rule of thumb, subtract one from the Xcode version to get the available SDK in nixpkgs.
+In some cases, you may have to use a non-default SDK. This can happen when a package requires APIs that are not present in the default SDK. For example, Metal Performance Shaders were added in macOS 12. If the default SDK is 11.3, then a package that requires Metal Performance Shaders will fail to build due to missing frameworks and symbols.
 
-  The `DEVELOPER_DIR` variable in the build environment has the path to the SDK in the build environment. The `SDKROOT` variable there contains a sysroot with the framework, header, and library paths. You can reference an SDK’s sysroot from Nix using the `sdkroot` attribute on the SDK package. Note that it is preferable to use `SDKROOT` because the latter will be resolved to the highest SDK version of any available to your derivation.
+To use a non-default SDK, add it to your derivation’s `buildInputs`. If your derivation needs a non-default SDK at build time (e.g., for a `depsBuildBuild` compiler), see the cross-compilation documentation for which input you should use.
```

Comment:
We should at least give an example or mention a package name here, since currently people will have to scroll a lot further to find out what they actually add to `buildInputs`.

Reply:
I added an example for the normal SDK override.